mongo/integration/client_side_encryption_prose_test.go

```diff
@@ -1140,7 +1140,9 @@ func TestClientSideEncryptionProse(t *testing.T) {
 					// Insert already encrypted data and verify that it is automatically decrypted by Next().
 					insertDocuments(mt, cpt.coll, []bson.Raw{testConfig.EncryptedDocument})
 					assert.True(mt, stream.Next(context.Background()), "expected Next to return true, got false")
-					gotDocument := stream.Current.Lookup("fullDocument").Document()
+					gotValue, err := stream.Current.LookupErr("fullDocument")
+					assert.Nil(mt, err, "did not find fullDocument in stream.Current: %v", stream.Current)
+					gotDocument := gotValue.Document()
```

Comment:
This change was motivated by [this test failure](https://evergreen.mongodb.com/lobster/evergreen/test/mongo_go_driver_tests_42_plus_zlib_zstd_support__version~6.0_os_ssl_40~windows_64_go_1_18_test_replicaset_auth_ssl_patch_766f8281c63d88369371f5b079e0381a75fc7953_63913f6561837d20741a5186_22_12_08_01_35_33/0/test/#bookmarks=0%2C20492&l=1&shareLine=14291) on a patch build:

```
panic: Call of bsoncore.Value.Document on invalid type [recovered]
	panic: Call of bsoncore.Value.Document on invalid type
goroutine 382036 [running]:
testing.tRunner.func1.2({0x1894440, 0xc00607b8d8})
	C:/golang/go1.18/src/testing/testing.go:1389 +0x24e
testing.tRunner.func1()
	C:/golang/go1.18/src/testing/testing.go:1392 +0x39f
panic({0x1894440, 0xc00607b8d8})
	C:/golang/go1.18/src/runtime/panic.go:838 +0x207
go.mongodb.org/mongo-driver/x/bsonx/bsoncore.Value.Document({0x22?, {0x0?, 0x1828f00?, 0xc00611ddc8?}})
	C:/data/mci/29507e88404966043ea412442a65263c/src/go.mongodb.org/mongo-driver/x/bsonx/bsoncore/value.go:443 +0x134
go.mongodb.org/mongo-driver/bson.RawValue.Document(...)
	C:/data/mci/29507e88404966043ea412442a65263c/src/go.mongodb.org/mongo-driver/bson/raw_value.go:145
go.mongodb.org/mongo-driver/mongo/integration.TestClientSideEncryptionProse.func9.2.1(0xc005651180)
	C:/data/mci/29507e88404966043ea412442a65263c/src/go.mongodb.org/mongo-driver/mongo/integration/client_side_encryption_prose_test.go:1143 +0x51a
go.mongodb.org/mongo-driver/mongo/integration/mtest.(*T).RunOpts.func1(0x15d2aa0?)
	C:/data/mci/29507e88404966043ea412442a65263c/src/go.mongodb.org/mongo-driver/mongo/integration/mtest/mongotest.go:263 +0x26f
testing.tRunner(0xc005b611e0, 0xc00e448400)
	C:/golang/go1.18/src/testing/testing.go:1439 +0x102
created by testing.(*T).Run
```

The test failure appears unrelated to the upgrade of libmongocrypt. More error information may help to diagnose in future failures.